mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -243,6 +245,19 @@ class ChiSqSelector @Since("2.1.0") () extends Serializable {
       case ChiSqSelector.FPR =>
         chiSqTestResult
           .filter { case (res, _) => res.pValue < alpha }
+      case ChiSqSelector.FDR =>
```

Comment:
It's definitely used -- you have to keep the original index in order to pass them to the model.